torch/_inductor/pattern_matcher.py

```diff
@@ -59,8 +64,33 @@
 NodeOrConstant = Union[Constant, torch.fx.Node]
 
 
+class SearchFn(Protocol):
+    __name__: str
+
+    def __call__(self, *args: Any, **kwargs: Any) -> Any:
+        ...
+
+
+class ReplaceFn(Protocol):
+    def __call__(self, *args: Any, **kwargs: Any) -> Any:
+        ...
+
+
+class TraceFn(Protocol):
+    def __call__(
+        self, fn: Union[SearchFn, ReplaceFn], *args: Any, **kwargs: Any
+    ) -> torch.fx.GraphModule:
+        ...
+
+
+T = TypeVar("T")
```

Comment:
Does this TypeVar need to be be public? Or should it be prefixed with an underscore?

Reply:
It's not public public - it's under _inductor so it's not in the public API.  On the other hand you're correct that it doesn't need to be visible outside this file. It seems like we generally have a lot of classes that fall into that category where as long as it's not in the public API we don't put the underscore.